### DISCUSSION TITLE
IPF Map direction still the same despite code changes

### DISCUSSION THREAD

**Discussion by bicer23** [mtex-toolbox/mtex#2168] [Ask Anything]
Dear reader, 

I have come across the issue, that despite changing the direction of the IPFmap, I still get the same map and key.
Below you can see the code I used for the map.
```Matlab
%%%% 
oM=ipfTSLKey(ebsd('Tin beta'));
oM.inversePoleFigureDirection = vector3d.X;
color=oM.orientation2color(ebsd('Tin beta').orientations);
ipfKey=ipfColorKey(ebsd('Tin beta'));

figure(1);
plot(ipfKey)
figure(2);
plot(ebsd('Tin beta'),ebsd('Tin beta').orientations);
```
I intended to change ipf direction by just changin vector3d.X into vector3d.Y/Z.

Thanks for your help

**Comment by kilir** [comment]
Hi,

you do not use the colorkey at all. Try:

```Matlab
plot(ebsd('Tin beta'),color);
%or directly
plot(ebsd('Tin beta'),oM.orientation2color(ebsd('Tin beta').orientations));

```
Cheers,

Rüdiger
